### PR TITLE
feat: add typescript in files redux

### PR DESCRIPTION
**Pull request by nikitsch**

### CODE REVIEW

File: src/redux/auth-reducer.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+// }
+
+let initialState = {
+  id: null as number | null,
+  email: null as string | null,
+  login: null as string | null,
+  isAuth: false,
+  isFetching: false,
+  captchaUrl: null as string | null // если null, значит капча не требуется
+};
```
Comment (GolDOragon) [inline src/redux/auth-reducer.ts:23]: Use `InitialStateType`

Comment (GolDOragon) [inline src/redux/auth-reducer.ts:23]: // If it's null, the captcha isn't required.

File: src/redux/auth-reducer.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+}
+export const setAuthUserData = (id: number | null, email: string | null, login: string | null, isAuth: boolean): SetAuthUserDataActionType => ({
+  type: SET_USER_DATA,
+  payload: { id, email, login, isAuth }
+});
+
+type GetCaptchaUrlSuccessActionType = {
+  type: typeof GET_CAPTCHA_URL_SUCCESS,
+  payload: { captchaUrl: string } //это то же описание типа
+}
```
Comment (GolDOragon) [inline src/redux/auth-reducer.ts:58]: it's also type description

File: src/redux/dialogs-reducer.ts [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
+}
+
+
 let initialState = {
   dialogsData: [
     { id: 1, name: "Mikita" },
     { id: 2, name: "Denis" },
     { id: 3, name: "Anastasia" },
-  ],
+  ] as Array<DialogType>,
```
Comment (GolDOragon) [inline src/redux/dialogs-reducer.ts:20]:
```ts
type InitialState = {
  dialogs: DialogType[],
  messages: MessageType[],
}

let initialState: InitialState = { ...  }
```

File: src/redux/sidebar-reducer.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+let initialState = {};
+type InitialStateType = typeof initialState;
```
Comment (GolDOragon) [inline src/redux/sidebar-reducer.ts:2]: format document

File: src/redux/users-reducer.ts [hunk 5]
```diff
@@ -10,16 +11,28 @@ const TOGGLE_IS_FETCHING = 'TOGGLE_IS_FETCHING';
 const TOGGLE_IS_FOLLOWING_PROGRESS = 'TOGGLE_IS_FOLLOWING_PROGRESS';
 // const PHOTO_URL = 'https://i.guim.co.uk/img/media/3cd8327353fe290e2bbfe195c1006b6e3f2f4be6/0_297_3415_2049/master/3415.jpg?width=1200&height=1200&quality=85&auto=format&fit=crop&s=34157ed80928d3d8722f5c9e69c52482';
 
+// type UserType = {
+//   items: UserItemsType
```
Comment (GolDOragon) [inline src/redux/users-reducer.ts:15]: use type

File: src/types/types.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export type ProfileType = {
+  userId: number
+  lookingForAJob: boolean
+  lookingForAJobDescription: string
+  fullName: string
+  contacts: ContactsType
+  photos: PhotosType
+}
```
Comment (GolDOragon) [inline src/types/types.ts:14]: PhotosType[] ?

**Review by GolDOragon — APPROVED** [review]

File: src/redux/auth-reducer.ts (hunk 1 above)
Comment (nikitsch) [reply src/redux/auth-reducer.ts:23]: on the lines [16-23] an alternative to the lines [7-14] for understanding

Comment (nikitsch) [reply src/redux/auth-reducer.ts:23]: remarks in English my brain can perceive as part of the code

File: src/redux/users-reducer.ts (hunk 5 above)
Comment (nikitsch) [reply src/redux/users-reducer.ts:15]: I import it from the folder 'type'

File: src/types/types.ts (hunk 6 above)
Comment (nikitsch) [reply src/types/types.ts:14]: as I understand it, JS allows ignoring this clarification

**Pull request merged**